Assets/Simon/Scripts/GameLoopManager.cs

```diff
@@ -0,0 +1,99 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class GameLoopManager : MonoBehaviour
+{
+    [SerializeField] private Button startGameButton;
+
+    private MiniGamePhases currentMiniGamePhases = MiniGamePhases.WatchingPhase;
+
+    private Sequence sequence;
+
+    private UIManager UImanager;
+
+    private ColorButtonEffectPlayer colorButtonEffectPlayer;
+
+    private int score = 0;
+
+
+    private void Awake()
+    {
+        startGameButton.onClick.RemoveAllListeners();
+        startGameButton.onClick.AddListener(StartGameLoop);
+
+        sequence = GameObject.FindObjectOfType<Sequence>();
+        UImanager = GameObject.FindObjectOfType<UIManager>();
+        colorButtonEffectPlayer = GameObject.FindObjectOfType<ColorButtonEffectPlayer>();
+    }
+
+    private void StartGameLoop()
+    {
+        StartCoroutine(StartGameLoopCoroutine());
+    }
+
+    private IEnumerator StartGameLoopCoroutine()
+    {
+        score = 0;
+        UImanager.ToggleGameOverUI(onOff: false);
+        sequence.CreateColorSequence(2);
+        yield return StartCoroutine(PlayColorSequenceCoroutine());
+    }
+
+    private IEnumerator PlayColorSequenceCoroutine()
+    {
+        currentMiniGamePhases = MiniGamePhases.WatchingPhase;
+        yield return StartCoroutine(colorButtonEffectPlayer.PlayButtonSequence(sequence.ColorSequence));
+        currentMiniGamePhases = MiniGamePhases.PlayingPhase;
+    }
+
+    public IEnumerator SelectColor(Colors color)
+    {
+        if (currentMiniGamePhases == MiniGamePhases.WatchingPhase)
+        {
+            yield break;
+        }
+
+        currentMiniGamePhases = MiniGamePhases.WatchingPhase;
+        yield return StartCoroutine(colorButtonEffectPlayer.PlayButton(color));
+        currentMiniGamePhases = MiniGamePhases.PlayingPhase;
+
+        if (sequence.CompareColorWithCurrentSequenceColor(color) == false && currentMiniGamePhases != MiniGamePhases.WatchingPhase)

```

Comment:
I removed currentMiniGamePhases != MiniGamePhases.WatchingPhase.

I still think adding "== false" makes thing clearer. I can remove it if you thing this is really weird.

Reply:
Doesn't really bother me but it is not intuitive at all tbh.

Like how in formal logic, negation notation is always _not [something]_ , rather than _[something] not_

![image](https://user-images.githubusercontent.com/61809127/132958387-ca7b81b0-ccf4-4bc0-89d7-d4398dcea94d.png)

I think that's why it is the convention.

Reply:
I see. I'll give it a try then. In that case, I reworked the method to include "IS" in its name: ColorIsSameAsCurrentSequenceColor. I think the name is even clearer that way.